src/parser/mod.rs

```diff
@@ -1141,13 +1141,26 @@ impl<'a, 'o> Parser<'a, 'o> {
             &self.options.extension.front_matter_delimiter,
         ) {
             if let Some((front_matter, rest)) = split_off_front_matter(s, delimiter) {
+                let lines = front_matter
+                    .as_bytes()
+                    .iter()
+                    .filter(|b| **b == b'\n')
+                    .count();
                 let node = self.add_child(
                     self.root,
                     NodeValue::FrontMatter(front_matter.to_string()),
                     1,
                 );
                 s = rest;
                 self.finalize(node).unwrap();
+                node.data.borrow_mut().sourcepos = Sourcepos {
+                    start: nodes::LineColumn { line: 1, column: 1 },
+                    end: nodes::LineColumn {
+                        line: lines,
+                        column: delimiter.chars().count(),
```

Comment:
This line I'm particularly unsure about. Are columns supposed to be bytes, characters, or actual columns (i.e. accounting for ZWJ, combining diacritics, etc.)?

Reply:
This question gave me pause too. Instead of me trying to infer it from some of my own code, let's try to get it from the source ([pun intended](https://www.youtube.com/watch?v=YbeZoV3Dcsc) (video unrelated)). If we ask [cmark](https://github.com/commonmark/cmark) for sourcepos of some text with minimally non-byte-oriented text …

```console
$ echo 'テスト test' | build/src/cmark --sourcepos
<p data-sourcepos="1:1-1:14">テスト test</p>
```

… we see it doesn't make any attempt to interpret UTF-8 whatsoever. This is a bit ¯\\\_(ツ)\_/¯ But it accords with our current behaviour, which makes sense given we modelled on it totally:

```console
$ echo 'テスト test' | cargo run -- --sourcepos
<p data-sourcepos="1:1-1:14">テスト test</p>
```

So, for consistency (and ease of your implementation), I'd say let's start with bytes, and I'll open an issue to think about improving this. There's a few places in `parser/mod.rs` where we actually use `self.column` and treat it as a byte count, and while I'm not sure it'd come into play here exactly (they're mostly around leading indent), it's another vote for bytes.